pkg/cluster/controllers/storage/tenant.go

```diff
@@ -96,10 +96,10 @@ func (m *MinIOController) getMinIOProperties(ctx context.Context, harborcluster
 	return storageSpec, nil
 }
 
-func (m *MinIOController) createSecretKeyRef(secretKey []byte, harborcluster *goharborv1.HarborCluster, minioInstance *minio.Tenant) *corev1.Secret {
+func (m *MinIOController) createSecretKeyRef(secretKey []byte, harborcluster *goharborv1.HarborCluster, minioInstance *miniov2.Tenant) *corev1.Secret {
 	s3KeySecret := &corev1.Secret{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: "v1",
+			APIVersion: "minio.min.io",
```

Comment:
Not `minio.min.io/v2`?

Reply:
Are you sure that the `corev1.Secret` should use the APIVersion `minio.min.io`? IMO there is some confusion here.
Once the [scheme is added to the runtime](https://github.com/goharbor/harbor-operator/blob/master/pkg/scheme/scheme.go). The `TypeMeta` field does not need to be specified.